storages/jsonl-storage/src/store_mut.rs

```diff
@@ -0,0 +1,176 @@
+            remove_file(data_path).map_storage_err()?;
+        }
+
+        let schema_path = self.schema_path(table_name);
+        if schema_path.exists() {
+            remove_file(schema_path).map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn append_data(&mut self, table_name: &str, rows: Vec<DataRow>) -> Result<()> {
+        let schema = self
+            .fetch_schema(table_name)?
+            .map_storage_err(JsonlStorageError::TableDoesNotExist)?;
+        let table_path = JsonlStorage::data_path(self, table_name);
+
+        let mut file = OpenOptions::new()
+            .write(true)
+            .append(true)
+            .open(table_path)
+            .map_storage_err()?;
+
+        let column_defs = schema.column_defs.unwrap_or_default();
+        let labels = column_defs
+            .iter()
+            .map(|column_def| column_def.name.as_str())
+            .collect::<Vec<_>>();
+
+        for row in rows {
+            let json_string = match row {
+                DataRow::Vec(values) => {
+                    let mut json_map = Map::new();
+                    for (key, value) in labels.iter().zip(values.into_iter()) {
+                        json_map.insert(key.to_string(), value.try_into()?);
+                    }
+
+                    JsonValue::Object(json_map).to_string()
+                }
+                DataRow::Map(hash_map) => {
+                    let mut json_map = Map::new();
+                    for (key, value) in hash_map {
+                        json_map.insert(key.to_string(), value.try_into()?);
+                    }
+
+                    JsonValue::Object(json_map).to_string()
+                }
+            };
+            writeln!(file, "{json_string}").map_storage_err()?;
+        }
+
+        Ok(())
+    }
+
+    async fn insert_data(&mut self, table_name: &str, mut rows: Vec<(Key, DataRow)>) -> Result<()> {
+        let prev_rows = self.scan_data(table_name)?;
+        rows.sort_by(|(key_a, _), (key_b, _)| {
+            key_a.to_cmp_be_bytes().cmp(&key_b.to_cmp_be_bytes())
+        });
+        let rows = rows.into_iter();
+
+        let sort_merge = SortMerge::new(prev_rows, rows);
+        let merged = sort_merge.collect::<Result<Vec<_>>>()?;
+
+        let table_path = self.data_path(table_name);
+        File::create(&table_path).map_storage_err()?;
+
+        self.append_data(table_name, merged).await
+    }
+
+    async fn delete_data(&mut self, table_name: &str, keys: Vec<Key>) -> Result<()> {
+        let prev_rows = self.scan_data(table_name)?;
+        let rows = prev_rows
+            .filter_map(|result| {
+                result
+                    .map(|(key, data_row)| {
+                        let preservable = !keys.iter().any(|target_key| target_key == &key);
+
+                        preservable.then_some(data_row)
+                    })
+                    .unwrap_or(None)
+            })
+            .collect::<Vec<_>>();
```

Comment:
```rust
             preservable.then_some(data_row)
         })
         .transpose()
 })
 .collect::<Result<Vec<_>>>()?;
```
It would be better to handle error explicitly.